core/src/test/scala/integration/kafka/api/KafkaAdminClientIntegrationTest.scala

```diff
@@ -0,0 +1,148 @@
+  def globalTimeout = Timeout.millis(120000);
+
+  val brokerCount = 3
+  lazy val serverConfig = new Properties
+
+  def createConfig() : util.HashMap[String, Object] = {
+    val config = new util.HashMap[String, Object]
+    config.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
+    config
+  }
+
+  def waitForTopics(client: AdminClient, expectedPresent: Seq[String], expectedMissing: Seq[String]): Unit = {
+    TestUtils.waitUntilTrue(() => {
+        val topics = client.listTopics().names().get()
+        expectedPresent.forall(topicName => topics.contains(topicName)) &&
+          expectedMissing.forall(topicName => !topics.contains(topicName))
+      }, "timed out waiting for topics")
+  }
+
+  def assertFutureExceptionTypeEquals(future: KafkaFuture[_], clazz: Class[_ <: Throwable]): Unit = {
+    try {
+      future.get()
+      fail("Expected CompletableFuture.get to return an exception")
+    } catch {
+      case e: ExecutionException =>
+        val cause = e.getCause()
+        assertTrue("Expected an exception of type " + clazz.getName + "; got type " +
+            cause.getClass().getName, clazz.isInstance(cause))
+    }
+  }
+
+  @Test
+  def testCloseAdministrativeClient(): Unit = {
+    val client = AdminClient.create(createConfig())
+    client.close()
+  }
+
+  @Test
+  def testListNodes(): Unit = {
+    val client = AdminClient.create(createConfig())
+    val brokerStrs = brokerList.split(",").toList.sorted
+    var nodeStrs : List[String] = null
+    do {
+      var nodes = client.describeCluster().nodes().get().asScala
+      nodeStrs = nodes.map ( node => s"${node.host}:${node.port}" ).toList.sorted
+    } while (nodeStrs.size < brokerStrs.size)
+    assertEquals(brokerStrs.mkString(","), nodeStrs.mkString(","))
+    client.close()
+  }
+
+  @Test
+  def testCreateDeleteTopics(): Unit = {
+    val client = AdminClient.create(createConfig())
+    val newTopics : List[NewTopic] = List(
+        new NewTopic("mytopic", 1, 1),
+        new NewTopic("mytopic2", 1, 1))
+    client.createTopics(newTopics.asJava,
+      new CreateTopicsOptions().validateOnly(true)).all().get()
+    waitForTopics(client, List(), List("mytopic", "mytopic2"))
+
+    client.createTopics(newTopics.asJava).all().get()
+    waitForTopics(client, List("mytopic", "mytopic2"), List())
+
+    val results = client.createTopics(newTopics.asJava).results()
+    assert(results.containsKey("mytopic"))
+    assertFutureExceptionTypeEquals(results.get("mytopic"), classOf[TopicExistsException])
+    assert(results.containsKey("mytopic2"))
+    assertFutureExceptionTypeEquals(results.get("mytopic2"), classOf[TopicExistsException])
+
+    val deleteTopics : Set[String] = Set("mytopic", "mytopic2")
+    client.deleteTopics(deleteTopics.asJava).all().get()
+    waitForTopics(client, List(), List("mytopic", "mytopic2"))
+
+    client.close()
+  }
+
+  @Test
+  def testGetAllBrokerVersions(): Unit = {
+    val client = AdminClient.create(createConfig())
```

Comment:
Seems like we always create an AdminClient and close it at the end. Maybe we can do that in `setUp` and `tearDown`? That way we don't leak resources if the test fails.